lib/byron/test/integration/Test/Integration/Byron/Scenario/API/Transactions.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NumericUnderscores #-}
+{-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module Test.Integration.Byron.Scenario.API.Transactions
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Wallet.Api.Link
+    ( Discriminate )
+import Cardano.Wallet.Api.Types
+    ( ApiByronWallet
+    , ApiT (..)
+    , ApiTransaction
+    , EncodeAddress (..)
+    , WalletStyle (..)
+    )
+import Cardano.Wallet.Primitive.AddressDerivation
+    ( NetworkDiscriminant (..) )
+import Cardano.Wallet.Primitive.Mnemonic
+    ( Mnemonic )
+import Cardano.Wallet.Primitive.Types
+    ( Address, Direction (..), TxStatus (..), WalletId )
+import Data.Aeson
+    ( FromJSON )
+import Data.Generics.Internal.VL.Lens
+    ( (^.) )
+import Data.Generics.Product.Typed
+    ( HasType )
+import Data.Quantity
+    ( Quantity (..) )
+import GHC.Generics
+    ( Generic )
+import Numeric.Natural
+    ( Natural )
+import Test.Hspec
+    ( SpecWith, describe, it, shouldBe )
+import Test.Integration.Framework.DSL
+    ( Context
+    , Headers (..)
+    , Payload (..)
+    , TxDescription (..)
+    , between
+    , eventually
+    , expectField
+    , expectResponseCode
+    , faucetAmt
+    , fixtureIcarusWallet
+    , fixturePassphrase
+    , fixtureRandomWallet
+    , icarusAddresses
+    , json
+    , randomAddresses
+    , request
+    , verify
+    )
+
+import qualified Cardano.Wallet.Api.Link as Link
+import qualified Network.HTTP.Types.Status as HTTP
+
+spec :: forall t n. (n ~ 'Mainnet) => SpecWith (Context t)
+spec = describe "BYRON_TXS" $ do
+    scenario_TRANS_CREATE_01 @'Byron fixtureRandomWallet (randomAddresses @n)
+    scenario_TRANS_CREATE_01 @'Byron fixtureIcarusWallet (icarusAddresses @n)
+
+--
+-- Scenarios
+--
+
+scenario_TRANS_CREATE_01
+    :: forall style t n mw wallet.
+        ( n ~ 'Mainnet
+        , Discriminate style
+        , HasType (ApiT WalletId) wallet
+        , FromJSON wallet
+        , Generic wallet
+        , Show wallet
+        )
+    => (Context t -> IO (wallet, Mnemonic mw))
+    -> (Mnemonic mw -> [Address])
+    -> SpecWith (Context t)
+scenario_TRANS_CREATE_01 fixtureWallet fixtureAddresses =
```

Comment:
*[question]*

Can I send from random wallet to icarus wallet and vice versa and expect it to work?

Reply:
Yes! Note that I have only ported a single scenario here, but I'll add a few more this morning, basically, most scenarios we already have in `API/Transactions`. While doing so, I am abstracting away the bits specific to wallets so that once done, we can simply use these functions directly in both cases (Jormungandr & cardano-node) without duplicating any code.